Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -112,21 +112,8 @@ bool getVideoSizeAndFormatFromCaps(const GstCaps* caps, WebCore::IntSize& size,
         return false;
     }
 
-    if (areEncryptedCaps(caps)) {
-        GstStructure* structure = gst_caps_get_structure(caps, 0);
-        format = GST_VIDEO_FORMAT_ENCODED;
-        stride = 0;
-        int width = 0, height = 0;
-        gst_structure_get_int(structure, "width", &width);
-        gst_structure_get_int(structure, "height", &height);
-        if (!gst_structure_get_fraction(structure, "pixel-aspect-ratio", &pixelAspectRatioNumerator, &pixelAspectRatioDenominator)) {
-            pixelAspectRatioNumerator = 1;
-            pixelAspectRatioDenominator = 1;
-        }
-
-        size.setWidth(width);
-        size.setHeight(height);
-    } else {
+    GstStructure* structure = gst_caps_get_structure(caps, 0);
+    if (!areEncryptedCaps(caps) && (!gst_structure_has_name(structure, "video/x-raw") || gst_structure_has_field(structure, "format"))) {
```

Comment:
Refactored this a bit to combine the encrypted caps codepaths with the no format codepaths.

Reply:
Why is `gst_video_info_from_caps()` not called for `video/x-raw` caps? This looks incorrect...

Reply:
Rephrasing... with this change, `gst_video_info_from_caps()` can actually be called for non raw caps, which triggers errors.

Reply:
> Rephrasing... with this change, `gst_video_info_from_caps()` can actually be called for non raw caps, which triggers errors.

I didn't change that behavior, it was always called from the else block before this change:
https://github.com/WebKit/WebKit/blob/d2b5507f6b8e81ec0d5bd4de2d3d583e3fd3d199/Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp#L129-L133